### PR TITLE
handle not found errors inside fileop

### PR DESCRIPTION
**Pull request by rbren**
I believe this fixes https://github.com/OpenDevin/OpenDevin/issues/1164

We were passing FileNotFound exceptions directly back to the LLM, which included absolute paths _outside_ the sandbox. This ensures that the error message only contains the filepath known to the LLM

### CODE REVIEW

**Review by enyst — APPROVED** [review]
Great catch!

**Comment by enyst** [comment]
While this looks to me like a needed change, I'm not sure it fixes the issue or all of the issue. I just noticed an absolute path in the observation of a successful FileReadAction. We might be creating some observations with an absolute path.

**Comment by rbren** [comment]
@enyst do you happen to know what the absolute path was?

If it's `/workspace/...` I think that's fine and kind of inevitable--that's the true absolute path in the sandbox

If it's `/opt/worksapce_base/...` or a directory on your local machine, that's an issue

**Comment by enyst** [comment]
> @enyst do you happen to know what the absolute path was?
> 
> If it's `/workspace/...` I think that's fine and kind of inevitable--that's the true absolute path in the sandbox
> 
> If it's `/opt/worksapce_base/...` or a directory on your local machine, that's an issue

'/Users/enyst/repos/...'. I'll try to replicate it.

File: opendevin/action/fileop.py [hunk 1]
```diff
@@ -38,21 +38,24 @@ class FileReadAction(ExecutableAction):
...
+                    else:
+                        all_lines = file.readlines()
+                        code_slice = all_lines[self.start:]
+                        code_view = ''.join(code_slice)
                 else:
                     all_lines = file.readlines()
-                    code_slice = all_lines[self.start:]
+                    num_lines = len(all_lines)
+                    begin = max(0, min(self.start, num_lines - 2))
+                    end = -1 if self.end > num_lines else max(begin + 1, self.end)
```
Comment (SmartManoj) [inline opendevin/action/fileop.py:54]: >>> a = [1,2,3]
>>> a[1:5]
[2, 3]
>>> a[1:-1]
[2]

Why setting end to -1 if self.end > num_lines?

**Pull request merged**